externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/sql/kyuubi/SparkDatasetHelper.scala

```diff
@@ -73,3 +79,53 @@ object SparkDatasetHelper {
     }
   }
 }
+
+trait CompressionCodec extends Serializable {
+  def outputStream(baos: ByteArrayOutputStream): OutputStream
+  def compress(byteArray: Array[Byte]): Array[Byte] = {
+    val baos = new ByteArrayOutputStream()
+    var os: OutputStream = null
+    try {
+      os = outputStream(baos)
+      os.write(byteArray)
+    } finally {
+      if (os != null) {
+        os.close()
+      }
+    }
+    baos.toByteArray
+  }
+}
+
+object CompressionCodecFactory {
+  def createCodec(tpe: String): CompressionCodec = {
+    tpe match {
+      case "lz4" => Lz4CompressionCodec
+      case "zstd" => ZstdCompressionCodec
+      case "gzip" => GZIPCompressionCodec
+      case _ => NoCompressionCodec
+    }
+  }
+}
+
+object NoCompressionCodec extends CompressionCodec {
+  override def outputStream(baos: ByteArrayOutputStream): OutputStream = {
+    throw new UnsupportedOperationException()
```

Comment:
If `no-compression` is not supported, why not just throw this exception in `CompressionCodecFactory`.`createCodec`?